.github/workflows/build-research.yml

```
@@ -0,0 +1,77 @@
+
+jobs:
+  build-and-test:
+    name: Build and Test Research
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
```

actionlint (1.7.9)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)